### ISSUE TITLE
Either CardDescription should have a names attr, in which case it should be populated and whoosh index should be updatable, or it should not in which case remove it from card_desription.py

### ISSUE THREAD

**Issue by vorpal-buildbot** [PennyDreadfulMTG/Penny-Dreadful-Tools#5933]
Reported on Discord by bakert#2193

**Comment by bakert** [comment]
I've removed 'names' but actually CardDescription is pretty inaccurate now. I think it's only passing mypy because of limits on static typing and the way we read field names in a loop from card.py instead of explicitly. We should make it resemble what we actually get from scryfall more removing the lies if not detailing exhaustively.

**Comment by silasary** [comment]
Scryfall Card objects are not the same as oracle Card objects.
We should not be reusing CardDescription for both kinds of card.

**Comment by bakert** [comment]
CardDescription is *only* for Scryfall (formerly mtgjson) card entries. The Card used in oracle comes from magic.models.card.

**Comment by bakert** [comment]
A scryfall card looks like this:

```
{
  "object": "card",
  "id": "ac709474-7790-483f-9ed5-ea7abccfce53",
  "oracle_id": "b2c6aa39-2d2a-459c-a555-fb48ba993373",
  "multiverse_ids": [],
  "name": "Island",
  "lang": "en",
  "released_at": "2019-02-16",
  "uri": "https://api.scryfall.com/cards/ac709474-7790-483f-9ed5-ea7abccfce53",
  "scryfall_uri": "https://scryfall.com/card/prw2/10/island?utm_source=api",
  "layout": "normal",
  "highres_image": true,
  "image_uris": {
    "small": "https://img.scryfall.com/cards/small/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.jpg?1548387660",
    "normal": "https://img.scryfall.com/cards/normal/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.jpg?1548387660",
    "large": "https://img.scryfall.com/cards/large/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.jpg?1548387660",
    "png": "https://img.scryfall.com/cards/png/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.png?1548387660",
    "art_crop": "https://img.scryfall.com/cards/art_crop/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.jpg?1548387660",
    "border_crop": "https://img.scryfall.com/cards/border_crop/front/a/c/ac709474-7790-483f-9ed5-ea7abccfce53.jpg?1548387660"
  },
  "mana_cost": "",
  "cmc": 0,
  "type_line": "Basic Land — Island",
  "oracle_text": "({T}: Add {U}.)",
  "colors": [],
  "color_identity": [
    "U"
  ],
  "legalities": {
    "standard": "legal",
    "future": "legal",
    "frontier": "legal",
    "modern": "legal",
    "legacy": "legal",
    "pauper": "legal",
    "vintage": "legal",
    "penny": "legal",
    "commander": "legal",
    "duel": "legal",
    "oldschool": "not_legal"
  },
  "games": [
    "paper"
  ],
  "reserved": false,
  "foil": true,
  "nonfoil": false,
  "oversized": false,
  "promo": true,
  "reprint": true,
  "set": "prw2",
  "set_name": "RNA Ravnica Weekend",
  "set_uri": "https://api.scryfall.com/sets/ee3a8eb6-0583-492b-8be5-265795d38038",
  "set_search_uri": "https://api.scryfall.com/cards/search?order=set&q=e%3Aprw2&unique=prints",
  "scryfall_set_uri": "https://scryfall.com/sets/prw2?utm_source=api",
  "rulings_uri": "https://api.scryfall.com/cards/ac709474-7790-483f-9ed5-ea7abccfce53/rulings",
  "prints_search_uri": "https://api.scryfall.com/cards/search?order=released&q=oracleid%3Ab2c6aa39-2d2a-459c-a555-fb48ba993373&unique=prints",
  "collector_number": "10",
  "digital": false,
  "rarity": "common",
  "watermark": "simic",
  "illustration_id": "157dfb2c-428b-41a6-aead-f8809291957d",
  "artist": "Yeong-Hao Han",
  "border_color": "black",
  "frame": "2015",
  "frame_effect": "",
  "full_art": false,
  "story_spotlight": false,
  "related_uris": {
    "tcgplayer_decks": "https://decks.tcgplayer.com/magic/deck/search?contains=Island&page=1&partner=Scryfall&utm_campaign=affiliate&utm_medium=scryfall&utm_source=scryfall",
    "edhrec": "http://edhrec.com/route/?cc=Island",
    "mtgtop8": "http://mtgtop8.com/search?MD_check=1&SB_check=1&cards=Island"
  }
}
```

**Comment by bakert** [comment]
This does not bear much resemblance to CardDescription.

**Comment by bakert** [comment]
```
CardDescription = TypedDict('CardDescription', {
    'all_parts': Optional[List[Any]],
    'card_faces': Optional[List[Any]],
    'cmc': int,
    'imageName': str,
    'layout': str,
    'legalities': List[Dict[str, Any]],
    'manaCost': str,
    'name': str,
    'oracle_text': str,
    'rarity': str,
    'set': str,
    'type_line': str,
    'types': List[str]
})
```

**Comment by bakert** [comment]
Scryfall entry for a meld card printing:

```
{
  "object": "card",
  "id": "b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa",
  "oracle_id": "f3e23d5e-bd88-4e7c-a3fb-db2a8cb05b22",
  "multiverse_ids": [
    439332
  ],
  "mtgo_id": 66429,
  "tcgplayer_id": 151642,
  "name": "Gisela, the Broken Blade",
  "lang": "en",
  "released_at": "2017-11-24",
  "uri": "https://api.scryfall.com/cards/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa",
  "scryfall_uri": "https://scryfall.com/card/v17/10a/gisela-the-broken-blade?utm_source=api",
  "layout": "meld",
  "highres_image": true,
  "image_uris": {
    "small": "https://img.scryfall.com/cards/small/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.jpg?1562702276",
    "normal": "https://img.scryfall.com/cards/normal/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.jpg?1562702276",
    "large": "https://img.scryfall.com/cards/large/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.jpg?1562702276",
    "png": "https://img.scryfall.com/cards/png/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.png?1562702276",
    "art_crop": "https://img.scryfall.com/cards/art_crop/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.jpg?1562702276",
    "border_crop": "https://img.scryfall.com/cards/border_crop/front/b/3/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa.jpg?1562702276"
  },
  "mana_cost": "{2}{W}{W}",
  "cmc": 4,
  "type_line": "Legendary Creature — Angel Horror",
  "oracle_text": "Flying, first strike, lifelink\nAt the beginning of your end step, if you both own and control Gisela, the Broken Blade and a creature named Bruna, the Fading Light, exile them, then meld them into Brisela, Voice of Nightmares.",
  "power": "4",
  "toughness": "3",
  "colors": [
    "W"
  ],
  "color_identity": [
    "W"
  ],
  "all_parts": [
    {
      "object": "related_card",
      "id": "2b45fb19-450d-40bd-91c7-b5ace4a77f2a",
      "component": "meld_part",
      "name": "Bruna, the Fading Light",
      "type_line": "Legendary Creature — Angel Horror",
      "uri": "https://api.scryfall.com/cards/2b45fb19-450d-40bd-91c7-b5ace4a77f2a"
    },
    {
      "object": "related_card",
      "id": "eba8bb03-6093-4e2b-99a2-a3fc5d8eb659",
      "component": "meld_result",
      "name": "Brisela, Voice of Nightmares",
      "type_line": "Legendary Creature — Eldrazi Angel",
      "uri": "https://api.scryfall.com/cards/eba8bb03-6093-4e2b-99a2-a3fc5d8eb659"
    },
    {
      "object": "related_card",
      "id": "b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa",
      "component": "meld_part",
      "name": "Gisela, the Broken Blade",
      "type_line": "Legendary Creature — Angel Horror",
      "uri": "https://api.scryfall.com/cards/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa"
    }
  ],
  "legalities": {
    "standard": "not_legal",
    "future": "not_legal",
    "modern": "legal",
    "legacy": "legal",
    "pauper": "not_legal",
    "vintage": "legal",
    "penny": "not_legal",
    "commander": "legal",
    "brawl": "not_legal",
    "duel": "legal",
    "oldschool": "not_legal"
  },
  "games": [
    "mtgo",
    "paper"
  ],
  "reserved": false,
  "foil": true,
  "nonfoil": false,
  "oversized": false,
  "promo": false,
  "reprint": true,
  "variation": false,
  "set": "v17",
  "set_name": "From the Vault: Transform",
  "set_type": "from_the_vault",
  "set_uri": "https://api.scryfall.com/sets/63c89a12-d115-4084-a4af-fceef40ca02f",
  "set_search_uri": "https://api.scryfall.com/cards/search?order=set&q=e%3Av17&unique=prints",
  "scryfall_set_uri": "https://scryfall.com/sets/v17?utm_source=api",
  "rulings_uri": "https://api.scryfall.com/cards/b3c2bd44-4d75-4f61-89c0-1f1ba4d59ffa/rulings",
  "prints_search_uri": "https://api.scryfall.com/cards/search?order=released&q=oracleid%3Af3e23d5e-bd88-4e7c-a3fb-db2a8cb05b22&unique=prints",
  "collector_number": "10a",
  "digital": false,
  "rarity": "mythic",
  "flavor_text": "She now hears only Emrakul's murmurs.",
  "illustration_id": "db5289ab-8aa4-412d-afd4-f7b7fef475fb",
  "card_back_id": "bd3a2b8a-87c2-484d-8da3-dc891f4ee9a2",
  "artist": "Clint Cearley",
  "border_color": "black",
  "frame": "2015",
  "full_art": false,
  "textless": false,
  "booster": false,
  "story_spotlight": false,
  "edhrec_rank": 1088,
  "related_uris": {
    "gatherer": "https://gatherer.wizards.com/Pages/Card/Details.aspx?multiverseid=439332",
    "tcgplayer_decks": "https://decks.tcgplayer.com/magic/deck/search?contains=Gisela%2C+the+Broken+Blade&page=1&partner=Scryfall&utm_campaign=affiliate&utm_medium=scryfall&utm_source=scryfall",
    "edhrec": "http://edhrec.com/route/?cc=Gisela%2C+the+Broken+Blade",
    "mtgtop8": "https://mtgtop8.com/search?MD_check=1&SB_check=1&cards=Gisela%2C+the+Broken+Blade"
  }
}
```

**Comment by bakert** [comment]
9327a650.
